modules/ROOT/partials/configuration/improved_custom_elements.adoc

```diff
@@ -0,0 +1,86 @@
+[[improved-custom-elements]]
+== Improved Custom Elements
+
+Enhancements have been made to the existing custom_elements option with the introduction of the `addCustomElements` API in {productname}. This improvement refines the structure of custom elements to accommodate more complex configurations, enabling better customization for users.
+
+* **New Custom Element Structure:** The custom_elements option and the `addCustomElements` API now support a more complex structure. This structure is defined by a record where the key represents the name of the element, and the value corresponds to a detailed specification.
+* **Attributes and Inheritance:** Elements can now inherit attributes and children from other specified elements using the extends property in the Custom Element Spec.
+* **Attribute and Children Specifications:** The Custom Element Spec includes properties such as `attributes` which lists attribute names, or `children` which lists valid child element names. The `children` property includes presets like `@global`, `@blocks`, `@phrasing`, and `@flow`.
```

Comment:
I wonder if this could be broken up into some structure where we explain them as separate things since there is a lot of details on each one. Not sure if a list, table or just blocks of text is the best. Maybe one should start using typescript records even. I think any JS developer should be able to read those. Maybe worth discussing.

Some rough idea on how to describe these things also noticed that I forgot about the `padEmpty` boolean:
* extends - Extends is a string that is the element name of what the new element should by default inherit it's properties from also using extends will add the element as a valid child to any spot that the extended element is valid in.
* attributes - Attributes is an array of strings where each string is either an attribute name or a attribute preset name like '@global'.
* children - Children is an array of strings where each string is either an element name or one of the fixed presets `@blocks` `@phrasing` or `@flow`.
* padEmpty - Boolean defaults to false. If set to true then the element will be added a &nbsp; when saved so that it always have some content.

Also the `@globals` preset is one of these might have forgotten to mention that.
https://github.com/tinymce/tinymce/blob/main/modules/tinymce/src/core/test/ts/atomic/schema/GlobalAttributesSetTest.ts

We can always circle back and fix this later to and just have it very shallow now in the first round. This is mostly for our self and future features like our custom components.

Reply:
Created a followup ticket for this, to be actioned post 7.0, so we can expand on the initial documentation for these options.